snapshot/src/main/java/io/zeebe/snapshots/broker/impl/FileBasedSnapshotStore.java

```diff
@@ -354,9 +369,12 @@ PersistedSnapshot newSnapshot(final FileBasedSnapshotMetadata metadata, final Pa
     }
 
     final var destination = buildSnapshotDirectory(metadata);
+    final var checksumFile = buildChecksumFilePath(metadata);
     moveToSnapshotDirectory(directory, destination);
+    writeChecksumFile(checksumFile, checksum);
```

Comment:
Write checksum first before moving the directory. It would help to detect partial moves in case of crashes.

Reply:
Good idea! I was thinking of using it as a marker file (as we do in 1.0) later, but that's even better.

Reply:
Ya. in 1.0 it is ok in which ever order, because if there is no checksum file we detect it as corrupted.